File: Javascript/library.js
// library.js
const fs = require('fs');

class Library {
  constructor() {
    this.books = [];
  }

  addBook(book) {
    this.books.push(book);
  }

  findBookByISBN(isbn) {
    return this.books.find(book => book.isbn === isbn);
  }

  findBooksByAuthor(author) {
    return this.books.filter(book => 
      book.author.toLowerCase().includes(author.toLowerCase())
    );
  }

  getTotalBooks() {
    return this.books.length;
  }

  loadBooksFromJson(jsonFilePath) {
    try {
      const jsonString = fs.readFileSync(jsonFilePath, 'utf8');
      const loadedBooks = JSON.parse(jsonString);
      loadedBooks.forEach(bookData => {
        this.addBook(new Book(bookData.isbn, bookData.title, bookData.author));
      });
      console.log(`Successfully loaded ${loadedBooks.length} books from JSON.`);
    } catch (error) {
      console.error(`Error loading books from JSON: ${error.message}`);
    }
  }
}

module.exports = { Book, Library };

File: Java/Library.java
// Library.java
import com.fasterxml.jackson.databind.ObjectMapper;
import java.io.File;
import java.io.IOException;
import java.util.ArrayList;
import java.util.List;
import java.util.stream.Collectors;

public class Library {
    private List<Book> books = new ArrayList<>();

    public void addBook(Book book) {
        books.add(book);
    }

    public Book findBookByISBN(String isbn) {
        return books.stream()
                .filter(book -> book.getIsbn().equals(isbn))
                .findFirst()
                .orElse(null);
    }

    public List<Book> findBooksByAuthor(String author) {
        return books.stream()
                .filter(book -> book.getAuthor().toLowerCase().contains(author.toLowerCase()))
                .collect(Collectors.toList());
    }

    public int getTotalBooks() {
        return books.size();
    }

    public void loadBooksFromJson(String jsonFilePath) {
        ObjectMapper objectMapper = new ObjectMapper();
        try {
            Book[] loadedBooks = objectMapper.readValue(new File(jsonFilePath), Book[].class);
            for (Book book : loadedBooks) {
                addBook(book);
            }
            System.out.println("Successfully loaded " + loadedBooks.length + " books from JSON.");
        } catch (IOException e) {
            System.err.println("Error loading books from JSON: " + e.getMessage());
        }
    }
}